From 782d0d9e42b5ced667590ebf086b72cfc0f40b31 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 30 Mar 2021 12:38:24 +0300
Subject: [PATCH] [SW-36473] boot info: removing nvme data

Instead of storing all the nvme data before reboot, only the relevant
data is stored in spsram in the fw info struct.
cleaning up the nvme data preservation code.

Change-Id: Ie22ad51b62db407dd487d0360d8b407e86a457bf
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/preboot.c b/preboot/preboot.c
index 91f59e9..b7ac680 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -263,12 +263,6 @@
 	device_id_vendor_id = regmap_readl(mmPCIE_DBI_DEVICE_ID_VENDOR_ID_REG);
 }
 
-/* leaving extra room for future data growth */
-#define DESC_BUFF_SIZE	(2 * sizeof(struct shm_nvme_mi_data))
-
-/* buffer size must be 32bit aligned */
-u8 boot_info_buff[ROUND_UP4(DESC_BUFF_SIZE)];
-
 static void preboot_process_last_boot_info(void)
 {
 	u32 boot_err = HL_FW_FIT_BOOT_ERR_NONE;
@@ -293,12 +287,6 @@
 
 		i2csi_set_boot_err(boot_err, true);
 	}
-
-	/* preserve nvme data descriptor if it's present in the sp sram */
-	if (!boot_info_get(HL_BOOT_INFO_NVME_STATUS, (void *)boot_info_buff,
-			   sizeof(boot_info_buff), &data_size))
-		boot_info_set(HL_BOOT_INFO_NVME_STATUS, (void *)boot_info_buff,
-			      data_size);
 }
 
 __weak void hl_dev_pll_init(void) {}
